tests/test_server.py

```diff
@@ -74,23 +83,31 @@ def test_server_parse_channel_data(server, rtm_start_fixture):
 def test_server_parse_user_data(server, rtm_start_fixture):
     server.parse_user_data(rtm_start_fixture["users"])
     # Find user by Name
-    userbyname = server.users.find('fakeuser')
-    assert type(userbyname) == User
-    assert userbyname == "fakeuser"
-    assert userbyname != "someotheruser"
+    user_by_name = server.users.find('fakeuser')
+    assert type(user_by_name) == User
+    assert user_by_name == "fakeuser"
+    assert user_by_name != "someotheruser"
     # Find user by ID
-    userbyid = server.users.find('U10CX1234')
-    assert type(userbyid) == User
-    assert userbyid == "fakeuser"
-    assert userbyid.email == 'fakeuser@example.com'
+    user_by_id = server.users.find('U10CX1234')
+    assert type(user_by_id) == User
+    assert user_by_id== "fakeuser"
+    assert user_by_id.email == 'fakeuser@example.com'
     # Don't find invalid user
-    userbyid = server.users.find('invaliduser')
-    assert type(userbyid) != User
+    user_by_id = server.users.find('invaliduser')
+    assert type(user_by_id) != User
 
 
 def test_server_cant_connect(server):
-    with pytest.raises(SlackLoginError):
-        reply = server.ping()
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.start",
+            status=403,
+            json={"ok": False}
+        )
+
+        with pytest.raises(SlackConnectionError) as e:
```

Comment:
i'm not sure what this is meant to do. how i'm currently reading this: "when a SlackConnectionError happens, ping the server". what does this accomplish other than swallowing the error?

Reply:
>In order to write assertions about raised exceptions, you can use pytest.raises as a context manager

https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions

Reply:
okay, but there's no assertion about the error in the body

Reply:
maybe i misunderstood. `server.ping()` causes the error, then wrapping it in `pytest.raises(ErrorName)` asserts that the error occurred and it is of that type?

Reply:
👍

Reply:
Changed the call to `rtm_connect` instead of ping, to more accurately reflect what it's testing